internal/plugin/grpc_client_plugin.go

```diff
@@ -99,6 +102,20 @@ func (gc *GrpcClient) Init(ctx context.Context, messagePipe bus.MessagePipeInter
 func (gc *GrpcClient) createConnection() error {
 	ctx := context.Background()
 
+	gc.instancesMutex.Lock()
+	defer gc.instancesMutex.Unlock()
+
+	if len(gc.instances) == 0 {
+		return fmt.Errorf("error waiting on instances to get updated")
+	}
+
+	gc.resourceMutex.Lock()
+	defer gc.resourceMutex.Unlock()
```

Comment:
Are these locks necessary seeing as this is part of the `Init()` function?

Reply:
The resourceMutex yes. I removed the instances mutex